src/BulletInverseDynamics/details/MultiBodyTreeImpl.cpp

```diff
@@ -155,12 +155,12 @@ int MultiBodyTree::MultiBodyImpl::generateIndexSets() {
 		} else {
 			if (-1 == parent) {
 				// multiple bodies are directly linked to the environment, ie, not a single root
-				error_message("building index sets parent(%zu)= -1 (multiple roots)\n", child);
+				error_message("building index sets parent(%d)= -1 (multiple roots)\n", static_cast<int>(child));
 			} else {
 				// should never happen
 				error_message(
-					"building index sets. parent_index[%zu]= %d, but m_parent_index.size()= %d\n",
-					child, parent, static_cast<int>(m_parent_index.size()));
+					"building index sets. parent_index[%d]= %d, but m_parent_index.size()= %d\n",
+					static_cast<int>(child), parent, static_cast<int>(m_parent_index.size()));
```

Comment:
`What I learned on the internet today was `%zu is the correct format to use for `size_t` (which appears to be what `idArrayIdx` is typedefed to in this context) but there might be some spotty support for `%zu` in Visual Studio.

Meanwhile, your compiler seems to think `%zu` expects `unsigned__int64` but your `idArrayIdx` is `unsigned__int32`.  This error probably won't show up for those who build in 64-bit mode.

So... what compiler are you using?